### PR TITLE
Update links to use https in README.md

### PR DESCRIPTION
**Pull request by thomaslevesque**
I noticed that the Markdown preview in VSCode didn't load the FakeItEasy logo because it was a non secure URL. So while I was at it, I updated all links to use https. (except sourcebrowser.io, which doesn't support https)

### CODE REVIEW

**Comment by blairconrad** [comment]
I like it, @thomaslevesque. But I wonder if while we're at it, we shouldn't update the other https…

```
+1 .editorconfig ???# EditorConfig is awesome: http://EditorConfig.org
+15 CONTRIBUTING.md Changes in functionality (new features, changed behavior, or bug fixes) should be described by [xBehave.net](http://xbehave.github.io/) acceptance tests in the `FakeItEasy.Specs` project. Doing so ensures that tests are written in language familiar to FakeItEasy's end users and are resilient to refactoring.
+19 CONTRIBUTING.md When writing integration or unit tests, use the [3A's pattern](http://defragdev.com/blog/?p=783) (Arrange, Act, Assert) with comments indicating each part.
+47 CONTRIBUTING.md The repository is configured to preserve line endings both on checkout and commit (the equivalent of `autocrlf` set to `false`). This means *you* are responsible for line endings. We recommend that you configure your diff viewer so that it does not ignore line endings. Any [wall of pink](http://www.hanselman.com/blog/YoureJustAnotherCarriageReturnLineFeedInTheWall.aspx) pull requests will not be accepted.
+51 CONTRIBUTING.md Try to keep lines of code no longer than 160 characters wide. This isn't a strict rule. Occasionally a line of code can be more readable if allowed to spill over slightly. A good way to remember this rule is to use the 'Column Guides' feature of the [Productivity Power Tools 2012](http://visualstudiogallery.msdn.microsoft.com/3a96a4dc-ba9c-4589-92c5-640e07332afd) extension for Visual Studio.
+79 CONTRIBUTING.md FakeItEasy uses the git branching model known as [GitFlow](http://nvie.com/posts/a-successful-git-branching-model/). As such, all development must be performed on a ["feature branch"](https://martinfowler.com/bliki/FeatureBranch.html) created from the main development branch, which is called `develop`. To submit a change:
+81 CONTRIBUTING.md 1. [Fork](http://help.github.com/forking/) the  [FakeItEasy repository](https://github.com/FakeItEasy/FakeItEasy/) on GitHub
+96 docs\argument-constraints.md [Comparing object instances with FakeItEasy](http://blog.jonathanchannon.com/2013/09/11/comparing-object-instances-with-fakeiteasy).
+77 docs\creating-fakes.md For those accustomed to [Moq](http://www.moqthis.com/) there is an
+89 docs\creating-fakes.md For an alternative look at migrating from Moq to FakeItEasy, see Daniel Marbach's blog post that talks about [Migration from Moq to FakeItEasy with Resharper Search Patterns](http://www.planetgeek.ch/2013/07/18/migration-from-moq-to-fakeiteasy-with-resharper-search-patterns/).
+3 docs\external-resources.md * [FakeItEasy courses on Pluralsight](http://www.pluralsight.com/tag/fakeiteasy)
+4 docs\external-resources.md * [The "fakeiteasy" tag on StackOverflow](http://stackoverflow.com/questions/tagged/fakeiteasy)
+5 docs\external-resources.md * [Mocking HttpContext with FakeItEasy](http://blog.jonathanchannon.com/2013/04/30/mocking-httpcontext-with-fake-it-easy/)
+6 docs\external-resources.md * [Migration from Moq to FakeItEasy with Resharper Search Patterns](http://www.planetgeek.ch/2013/07/18/migration-from-moq-to-fakeiteasy-with-resharper-search-patterns/)
+3 docs\snippets-for-fakeiteasy-in-sidewaffle.md The [SideWaffle](http://sidewaffle.com/) extension adds a bunch of useful Snippets, Project- and Item Templates to Visual Studio, which you download through "Extensions and Updates..." under Tools in Visual Studio.
+6 docs\what-can-be-faked.md [Castle DynamicProxy](http://www.castleproject.org/projects/dynamicproxy/)
+13 docs\why-was-fakeiteasy-created.md "[Are fakes better than Mocks?](http://stackoverflow.com/questions/4001101/are-fakes-better-than-mocks)"
+11 src\FakeItEasy.Analyzer.nuspec     <projectUrl>http://fakeiteasy.github.io/</projectUrl>
+12 src\FakeItEasy.Analyzer.nuspec     <iconUrl>http://fakeiteasy.github.io/img/fakeiteasy_logo_256_square_white.png</iconUrl>
+1 src\FakeItEasy.Dictionary.xml <!-- NOTE (adamralph): For more info see: http://msdn.microsoft.com/en-us/library/bb514188.aspx -->
+9 src\FakeItEasy\IHideObjectMembers.cs     /// easier to read. Found in the source of Autofac: <see cref="!:http://code.google.com/p/autofac/"/>
+11 src\FakeItEasy\IHideObjectMembers.cs     /// <see cref="!:http://www.clariusconsulting.net/blogs/kzu/archive/2008/03/10/58301.aspx"/>
+19 src\FakeItEasy\Core\FakeCallEqualityComparer.cs         // NOTE (adamralph): based on http://stackoverflow.com/a/263416/49241
```

We can make this another issue, if you prefer.

**Comment by thomaslevesque** [comment]
> But I wonder if while we're at it, we shouldn't update the other https…

Good point. I'll do it.

File: src/FakeItEasy/IHideObjectMembers.cs [hunk 1]
```diff
@@ -6,9 +6,8 @@ namespace FakeItEasy
 
     /// <summary>
     /// Hides standard Object members to make fluent interfaces
-    /// easier to read. Found in the source of Autofac: <see cref="!:http://code.google.com/p/autofac/"/>
+    /// easier to read. Found in the source of Autofac: <see cref="!:https://code.google.com/p/autofac/"/>
     /// Based on blog post here:
-    /// <see cref="!:http://www.clariusconsulting.net/blogs/kzu/archive/2008/03/10/58301.aspx"/>
```
Comment (thomaslevesque) [inline src/FakeItEasy/IHideObjectMembers.cs]: Link is broken. I couldn't find another article explaining this.

File: src/FakeItEasy.Dictionary.xml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<!-- NOTE (adamralph): For more info see: http://msdn.microsoft.com/en-us/library/bb514188.aspx -->
+<!-- NOTE (adamralph): For more info see: https://docs.microsoft.com/en-us/visualstudio/code-quality/how-to-customize-the-code-analysis-dictionary -->
```
Comment (thomaslevesque) [inline src/FakeItEasy.Dictionary.xml]: The link redirects there, so I put the new URL

File: src/FakeItEasy/IHideObjectMembers.cs [hunk 3]
```diff
@@ -6,9 +6,7 @@ namespace FakeItEasy
 
     /// <summary>
     /// Hides standard Object members to make fluent interfaces
-    /// easier to read. Found in the source of Autofac: <see cref="!:http://code.google.com/p/autofac/"/>
-    /// Based on blog post here:
-    /// <see cref="!:http://www.clariusconsulting.net/blogs/kzu/archive/2008/03/10/58301.aspx"/>
```
Comment (thomaslevesque) [inline src/FakeItEasy/IHideObjectMembers.cs]: Link is broken, and I couldn't find another article explaining this.

Comment (blairconrad) [reply src/FakeItEasy/IHideObjectMembers.cs]: http://blogs.clariusconsulting.net/kzu/how-to-hide-system-object-members-from-your-interfaces/ ?

Comment (thomaslevesque) [reply src/FakeItEasy/IHideObjectMembers.cs]: Doh! Your Google-fu is better than mine. Still no HTTPS, though. I'll amend.

Comment (blairconrad) [reply src/FakeItEasy/IHideObjectMembers.cs]: > Your Google-fu is better than mine

Today, perhaps.

**Comment by blairconrad** [comment]
Thanks, @thomaslevesque!

**Comment by thomaslevesque** [comment]
Thanks for the merge!

**Comment by blairconrad** [comment]
This change has been released as part of [FakeItEasy 4.9.1](https://github.com/FakeItEasy/FakeItEasy/releases/4.9.1).

**Pull request merged**